code_examples/model_registration_pipeline_example.py

```diff
@@ -0,0 +1,181 @@
+                "pipeline_namespace": pipeline_namespace,
+                "kfp_component_name": "register-model-to-kubeflow-registry"
+            },
+            # Best practices for cross-referencing model source (from KEP):
+            model_source_id=pipeline_run_id,           # ID of the KFP pipeline run
+            model_source_name=pipeline_name,           # Name of the KFP pipeline
+            model_source_class="pipelinerun",          # KFP-specific identifier for run
+            model_source_kind="kfp",                   # KFP-specific identifier for source system
+            model_source_group=pipeline_namespace,     # Namespace where the pipeline ran
+        )
+        print(f"Registered Model ID: {registered_model_and_version.id}, Name: {registered_model_and_version.name}")
+        # The object returned might be a RegisteredModel or ModelVersion depending on client version.
+        # Assuming it's RegisteredModel, we can infer the version was created.
+
+        print(f"Successfully registered model and version in Model Registry.")
+        
+        # --- NEW: Set metadata on the KFP output model artifact ---
+        # Write some dummy content to the output model path
+        with open(output_model.path, 'w') as f:
+            f.write(f"This is a fake model artifact for {model_name} version {model_version_name}.")
+        
+        # Set metadata on the KFP output model artifact
+        output_model.metadata["registered_model"] = {
+            "modelName": registered_model_and_version.name,
+            "versionName": model_version_name, # Use the version name passed to component
+            "modelID": registered_model_and_version.id,
+            # In a real scenario, you'd get the actual modelVersionId from the API response
+            # For now, we'll use a placeholder or query it if needed.
+            "versionID": "placeholder-version-id", # Model Registry client doesn't return version ID directly from register_model
+            "modelRegistryURL": f"http://localhost:{MODEL_REGISTRY_PORT}/models/{registered_model_and_version.id}/versions/{model_version_name}",
+            "modelRegistryAPIEndpoint": MODEL_REGISTRY_SERVER_ADDRESS
+        }
+        print(f"KFP output model artifact metadata set: {output_model.metadata}")
+
+        # Return the registered model ID as a string output
+        return registered_model_and_version.id
+
+    except Exception as e:
+        print(f"An error occurred during model registration: {e}")
+        raise # Re-raise the exception to mark the task as failed
+
+@dsl.pipeline(
+    name="Model Registration Pipeline",
+    description="A KFP pipeline to demonstrate registering a model in Kubeflow Model Registry."
+)
+def model_registration_pipeline(
+    model_name: str = "MyFakeModel",
+    model_version_name: str = "v1.0.0-blog-post", # Unique version for blog post example
+    model_artifact_uri: str = "s3://my-model-bucket/fake/v1.0.0/model.pkl",
```

Comment:
Doesn't this come from `output_model.uri`? We don't need a pipeline input parameter.